feat: add sections about testing (#100)
GreptimeTeam/docs: docs/developer-guide/tests/integration-test.md

# Integration Test

## Introduction

> An integration test is also written with Rust test harness (`#[test]`). But unlike unit test they are placed separately under

Reviewer: ```suggestion
Integration testing is written with Rust test harness (`#[test]`), unlike unit testing, they are placed separately 
```

[here](https://github.com/GreptimeTeam/greptimedb/tree/develop/tests-integration).

> It's used to cover scenarios involving multiple components. One typical case is HTTP/gRPC-related features. You can check

Reviewer: ```suggestion
It covers scenarios involving multiple components, in which one typical case is HTTP/gRPC-related features. You can check
```
